core/controllers/story_editor.py

```diff
@@ -69,17 +71,23 @@ def get(self, topic_id, story_id):
         story_domain.Story.require_valid_story_id(story_id)
         topic_domain.Topic.require_valid_topic_id(topic_id)
 
-        story = story_services.get_story_by_id(story_id, strict=False)
+        story = story_fetchers.get_story_by_id(story_id, strict=False)
         if story is None:
             raise self.PageNotFoundException
 
         topic = topic_services.get_topic_by_id(topic_id, strict=False)
-        if topic is None or story_id not in topic.canonical_story_ids:
+        canonical_story_ids = topic.get_canonical_story_ids()
+        if topic is None or story_id not in canonical_story_ids:
```

Comment:
Here and elsewhere: could topic be None if you call a method on it in the previous line? You'll probably need to check existence earlier.

Make sure backend tests are comprehensive to cover these cases.

Reply:
Done